src/components/icon/library.ts

```diff
@@ -4,10 +4,12 @@ import type SlIcon from '../icon/icon';
 
 export type IconLibraryResolver = (name: string) => string;
 export type IconLibraryMutator = (svg: SVGElement) => void;
+export type IconLibrarySymbolSprite = 'external' | 'inline';
```

Comment:
Made it a string in case we want to support different spriting mechanisms like inlining.

Reply:
Hmmm...I'm kind of curious what would warrant needing to define inline vs external if the resolver handles how you construct the href. This feels like a boolean, but I'm happy to be wrong as well.

Reply:
Yeah, the more I thought about it, the less the inline version made sense.

The only use case is when for some reason, you need to control the cache headers of that request, and so we want to avoid the browser making an N request, but I really start questioning if is worth the trouble.

Happy to keep it a boolean for now as we can always change it later.